sdk/src/types/block/core/wrapper.rs

```diff
@@ -0,0 +1,360 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::mem::size_of;
+
+use crypto::hashes::{blake2b::Blake2b256, Digest};
+use packable::{
+    error::{UnexpectedEOF, UnpackError, UnpackErrorExt},
+    packer::{Packer, SlicePacker},
+    unpacker::{CounterUnpacker, SliceUnpacker, Unpacker},
+    Packable, PackableExt,
+};
+
+use crate::types::block::{
+    block_id::{BlockHash, BlockId},
+    core::{BasicBlock, ValidationBlock},
+    protocol::ProtocolParameters,
+    signature::{Ed25519Signature, Signature},
+    slot::{SlotCommitmentId, SlotIndex},
+    Block, Error, IssuerId,
+};
+
+/// Represent the object that nodes gossip around the network.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct BlockWrapper {
+    /// Protocol parameters of the network to which this block belongs.
+    // TODO Not fully sure this is worth the small UX improvement, needs to be discussed more.
+    protocol_params: ProtocolParameters,
+    /// The time at which the block was issued. It is a Unix timestamp in nanoseconds.
+    issuing_time: u64,
+    /// The identifier of the slot to which this block commits.
+    slot_commitment_id: SlotCommitmentId,
+    /// The slot index of the latest finalized slot.
+    latest_finalized_slot: SlotIndex,
+    /// The identifier of the account that issued this block.
+    issuer_id: IssuerId,
+    /// The inner block.
+    block: Block,
+    /// The block signature, used to validate issuance capabilities.
+    signature: Signature,
+}
+
+impl BlockWrapper {
+    /// The minimum number of bytes in a block.
+    pub const LENGTH_MIN: usize = 46;
+    /// The maximum number of bytes in a block.
+    pub const LENGTH_MAX: usize = 32768;
+    /// The length of the block header.
+    pub const HEADER_LENGTH: usize = size_of::<u8>()
```

Comment:
Was this to avoid a dynamic allocation ? Bit error prone

Reply:
Well it's const data. We could have a const fn instead I guess

Reply:
Yeah, just a bit error prone, which is why maybe a BlockHeader would be interesting so we take the size_of of this and we don't have to do an addition and potentially having it out of sync

Reply:
not a bad idea, I'm fine with that

Reply:
Opened an issue